backend/presentation/tests.py

```diff
@@ -0,0 +1,325 @@
+        )
+
+    def test_patch_presentation_detail_as_admin(self):
+        self.client.force_authenticate(user=self.admin)
+        response = self.client.patch(
+            path=self.detail_url, data=self.updated_presentation_data, format="json"
+        )
+        self.presentation_1.refresh_from_db()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            response.data, PresentationSerializer(self.presentation_1).data
+        )
+
+    def test_delete_presentation_without_auth(self):
+        response = self.client.delete(path=self.detail_url)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_delete_presentation_as_not_owner(self):
+        self.client.force_authenticate(user=self.user_2)
+        response = self.client.delete(path=self.detail_url)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_delete_presentation_as_owner(self):
+        self.client.force_authenticate(user=self.user_1)
+        response = self.client.delete(path=self.detail_url)
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(Presentation.objects.all().count(), 0)
+
+    def test_delete_presentation_as_admin(self):
+        self.client.force_authenticate(user=self.admin)
+        response = self.client.delete(path=self.detail_url)
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(Presentation.objects.all().count(), 0)
+
+
+class TestTagViewSet(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.admin = User.objects.create_user(
+            username="admin",
+            password="password123",
+            email="admin@mail.com",
+            first_name="admin",
+            last_name="admin",
+            is_superuser=True,
+        )
+        cls.user_1 = User.objects.create_user(
+            username="user_1",
+            password="password123",
+            email="user_1@mail.com",
+            first_name="user",
+            last_name="test_1",
+        )
+        cls.tag_1 = Tag.objects.create(name="Test_tag_1")
+        cls.tag_2 = Tag.objects.create(name="Test_tag_2")
+        cls.new_tag = {"name": "Test_tag_3"}
+        cls.updated_tag = {"name": "Test_updated_tag"}
+        cls.list_url = reverse("presentation:tag-list")
+        cls.detail_url = reverse("presentation:tag-detail", kwargs={"pk": cls.tag_1.pk})
+
+    def test_get_tag_list_without_auth(self):
+        response = self.client.get(path=self.list_url)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_get_tag_list(self):
+        self.client.force_authenticate(user=self.user_1)
+        response = self.client.get(path=self.list_url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 2)
```

Comment:
Now that I see that, maybe you could consider adding paging?

Reply:
Paging will be added in next task.